Detectors/DCS/testWorkflow/DCSDataProcessorSpec.h

```diff
@@ -0,0 +1,244 @@
+    LOG(INFO) << "Number of DPs received = " << nDPs;
+    std::unordered_map<DPID, DPVAL> dcsmap;
+    DPCOM dptmp;
+    LOG(INFO) << "TF: " << tfid << " -->  building unordered_map...";
+    s.Start();
+    for (int i = 0; i < nDPs; i++) {
+      memcpy(&dptmp, rawchar + i * sizeof(DPCOM), sizeof(DPCOM));
+      dcsmap[dptmp.id] = dptmp.data;
+      LOG(DEBUG) << "Reading from generator: i = " << i << ", DPCOM = " << dptmp;
+      LOG(DEBUG) << "Reading from generator: i = " << i << ", DPID = " << dptmp.id;
+    }
+    s.Stop();
+    LOG(INFO) << "TF: " << tfid << " -->  ...unordered_map built = " << s.RealTime() << ", cpuTime = " << s.CpuTime();
+
+    // delta map
+    const auto* dhDelta = o2::header::get<o2::header::DataHeader*>(pc.inputs().get("inputDelta").header);
+    auto szDelta = dhDelta->payloadSize;
+    int nDPsDelta = szDelta / sizeof(DPCOM);
+    LOG(INFO) << "Number of DPs received (delta map) = " << nDPsDelta;
+    std::unordered_map<DPID, DPVAL> dcsmapDelta;
+    LOG(INFO) << "TF: " << tfid << " -->  building (delta) unordered_map...";
+    s.Start();
+    for (int i = 0; i < nDPsDelta; i++) {
+      memcpy(&dptmp, rawcharDelta + i * sizeof(DPCOM), sizeof(DPCOM));
+      dcsmapDelta[dptmp.id] = dptmp.data;
+      LOG(DEBUG) << "Reading from generator: i = " << i << ", DPCOM = " << dptmp;
+      LOG(DEBUG) << "Reading from generator: i = " << i << ", DPID = " << dptmp.id;
+    }
+    s.Stop();
+    LOG(INFO) << "TF: " << tfid << " -->  ...unordered_map (delta) built = " << s.RealTime() << ", cpuTime = " << s.CpuTime();
+
+    if (tfid % 6000 == 0) {
+      //for (int idet = 0; idet < detVect.size(); idet++) {
+      for (int idet = 0; idet < 1; idet++) { // for now I test only 1 DCS processor
+        LOG(INFO) << "TF: " << tfid << " -->  starting processing...";
+        s.Start();
+        mDCSprocVect[idet].processMap(dcsmap, false);
+        s.Stop();
+        LOG(INFO) << "TF: " << tfid << " -->  ...processing done: realTime = " << s.RealTime() << ", cpuTime = " << s.CpuTime();
+      }
+    } else {
+      if (mProcessFullDeltaMap) {
+        //for (int idet = 0; idet < detVect.size(); idet++) {
+        for (int idet = 0; idet < 1; idet++) { // for now I test only 1 DCS processor
+          LOG(INFO) << "TF: " << tfid << " -->  starting (delta) processing...";
+          s.Start();
+          mDCSprocVect[idet].processMap(dcsmapDelta, true);
+          s.Stop();
+          LOG(INFO) << "TF: " << tfid << " -->  ...processing (delta) done: realTime = " << s.RealTime() << ", cpuTime = " << s.CpuTime();
+        }
+      } else {
+
+        // processing per DP found in the map, to be done in case of a delta map
+
+        bool resetStopwatch = true;
+        LOG(INFO) << "TF: " << tfid << " -->  starting (delta) processing for detector...";
+        for (const auto& dpcom : dcsmapDelta) {
+          std::vector<int> detVect = mDetectorAlias[dpcom.first];
+          //for (int idet = 0; idet < detVect.size(); idet++) {
```

Comment:
should not you indeed loop over entries of ``detVect[i]`` ?

Reply:
Yes, it was just for test, as I had defined 2 detectors, but I have the processor for one only. Now I deinfe only 1, so I can use the correct loop.